site/docs/features/index.md

```diff
@@ -1,107 +1,327 @@
+Apache [Hive](https://hive.apache.org)[^2] or Apache
+[Spark](https://spark.apache.org)[^2]. Your existing jobs can easily integrate
+Nessie without any production code changes, it's a simple configuration change.
+
+## git 101
+
+> *"Git is a free and open source distributed version control system designed to 
+handle everything from small to very large projects with speed and efficiency"*
+(cite from [git-scm.com](https://git-scm.com/))
+
+git maintains the history or all changes of a software project from the very
+first *commit* until the current state.
+
+git is used by humans, i.e. developers.
+
+Many of the concepts of git for source code are implemented by Nessie for all
+the data your data lake. It would be rather confusing to explain all git
+concepts here and then outline the differences in the next chapter.
+
+## Working with data in Nessie
+
+Each individual state in Nessie is defined by a *Nessie commit*.
+Each *commit* in Nessie, except the very first one, has references to its
+predecessors, the previous versions of the data.
+
+Each *Nessie commit* also "knows" about the data files in your data lake, which
+represent the state of all data in all tables.
+
+The following example illustrates that our *current commit* adds a 3rd data file.
+The other two data files 1+2 have been added by *previous commit*.
 ```
-$ create t1
-...
-$ insert 2 records into t1
-...
-$ create t2
-...
-$ insert 2 records into t2
-...
+ +-------------------+       +-------------------------+
+ |  previous commit  | --<-- |     current commit      |
+ +-------------------+       +-------------------------+
+     |         |                 |        |        |
+   (add)     (add)               |        |      (add)
+     |         |                 |        |        |
+  +------+  +------+          +------+ +------+ +------+
+  | data |  | data |          | data | | data | | data |
+  | file |  | file |          | file | | file | | file |
+  | #1   |  | #2   |          | #1   | | #2   | | #3   |
+  |     _|  |     _|          |     _| |     _| |     _|
+  |  __/    |  __/            |  __/   |  __/   |  __/  
+  |_/       |_/               |_/      |_/      |_/     
 ```
+In "relational SQL" you can think of the following sequence of SQL statements:
+```SQL
+BEGIN TRANSACTION;
+  -- The data for data file #1
+  INSERT INTO table_one (...) VALUES (...);
+  -- The data for data file #2
+  INSERT INTO other_table (...) VALUES (...);
+-- creates our "previous commit"
+COMMIT TRANSACTION;
+
+BEGIN TRANSACTION;
+  -- Data added to 'table_one' will "land" in a new data file #3, because
+  -- data files are immutable.
+  INSERT INTO table_one (...) VALUES (...);
+-- Creates our "current commit"
+COMMIT TRANSACTION; 
+```
+
+Each commit is identified by a sequence of hexadecimal characters like
+`2898591840e992ec5a7d5c811c58c8b42a8e0d0914f86a37badbeedeadaffe`, which is not
+easy to read and remember for us humans.
```

Comment:
perhaps add a technical footnote pointing out that its a SHA hash? Dunno if its mentioned elsewhere but the first thing I thought to myself after reading this was 'why'.